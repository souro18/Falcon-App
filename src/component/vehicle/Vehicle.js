import React from 'react'
import './vehicle.scss'

const Vehicle = props => {
    return (
        <div className="vehicles form-group" >
            {props.vehicle.map(v =>
                <div className='vehicle' key={v.name} data-testid="vehicle">
                    <input type='radio' value={v.name} name={'vehicle' + props.id} key={props.name} disabled={v.total_no === 0 || props.distance === 0 || props.distance > v.max_distance} onClick={e => props.onInput(e.target.value)} />
                    <label>
                        <span>{v.name}</span><br/>
                        <span>No. of vehicle: {v.total_no}</span><br />
                        <span>Max Distance: {v.max_distance}</span>
                    </label>
                </div>
            )}
        </div>
    )
}
export default Vehicle;