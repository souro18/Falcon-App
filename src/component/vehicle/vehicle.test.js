import React from 'react';
import {render , cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'

import Vehicle from './Vehicle'

afterEach(cleanup)
const vehicle = [
    {
        name: "pod",
        max_distance : 200,
        total_no: 2
    },
    {
        name: "sapce ship",
        max_distance : 600,
        total_no: 1
    }
]
it("renders vehicle" , ()=> {
    const {asFragment} = render(<Vehicle vehicle={vehicle} onInput={ e => {}} id={1} distance={300}/>)
    expect(asFragment()).toMatchSnapshot()
})
it("renders correct number of vehicle" , ()=> {
    const {getAllByTestId } = render(<Vehicle vehicle={vehicle} onInput={ e => {}} id={1} distance={300}/>)
    expect(getAllByTestId('vehicle').length).toBe(vehicle.length)
})