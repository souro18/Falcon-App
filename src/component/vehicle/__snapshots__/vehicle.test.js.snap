// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders vehicle 1`] = `
<DocumentFragment>
  <div
    class="vehicles form-group"
  >
    <div
      class="vehicle"
      data-testid="vehicle"
    >
      <input
        disabled=""
        name="vehicle1"
        type="radio"
        value="pod"
      />
      <label>
        <span>
          pod
        </span>
        <br />
        <span>
          No. of vehicle: 2
        </span>
        <br />
        <span>
          Max Distance: 200
        </span>
      </label>
    </div>
    <div
      class="vehicle"
      data-testid="vehicle"
    >
      <input
        name="vehicle1"
        type="radio"
        value="sapce ship"
      />
      <label>
        <span>
          sapce ship
        </span>
        <br />
        <span>
          No. of vehicle: 1
        </span>
        <br />
        <span>
          Max Distance: 600
        </span>
      </label>
    </div>
  </div>
</DocumentFragment>
`;
